simtools/io_handler.py

```diff
@@ -1,98 +1,139 @@
-
-    Parameters
-    ----------
-    filesLocation: str, or Path
-        Main location of the output files.
-    label: str
-        Instance label.
-    dirType: str
-        Name of the subdirectory (ray-tracing, model etc)
-    test: bool
-        If true, return test output location
-
-    Returns
-    -------
-    Path
-    """
-    _logger = logging.getLogger(__name__)
-
-    if test:
-        outputDirectoryPrefix = Path(cfg.get("outputLocation")).joinpath("test-output")
-    else:
-        if filesLocation is None:
-            filesLocation = cfg.get("outputLocation")
-        outputDirectoryPrefix = Path(filesLocation).joinpath("simtools-output")
-
-    today = datetime.date.today()
-    labelDir = label if label is not None else "d-" + str(today)
-    path = outputDirectoryPrefix.joinpath(labelDir)
-    if dirType is not None:
-        path = path.joinpath(dirType)
-    try:
-        path.mkdir(parents=True, exist_ok=True)
-    except FileNotFoundError:
-        _logger.error("Error creating directory {}".format(str(path)))
-        raise
-
-    return path.absolute()
-
-
-def getOutputFile(fileName, label=None, dirType=None, test=False):
+class IOHandlerSingleton(type):
     """
-    Get path of an output file.
-
-    Parameters
-    ----------
-    filesName: str
-        File name.
-    label: str
-        Instance label.
-    dirType: str
-        Name of the subdirectory (ray-tracing, model etc)
-    test: bool
-        If true, return test output location
-
-    Returns
-    -------
-    Path
+    Singleton base class
+
     """
-    return getOutputDirectory(label=label, dirType=dirType, test=test).joinpath(fileName).absolute()
 
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(IOHandlerSingleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
 
-def getInputDataFile(parentDir=None, fileName=None, test=False):
+
+class IOHandler(metaclass=IOHandlerSingleton):
     """
-    Get path of a data file, using the dataLocation taken from the config file.
-
-    Parameters
-    ----------
-    parentDir: str
-        Parent directory of the file.
-    filesName: str
-        File name.
-    test: bool
-        If true, return test resources location
-
-    Returns
-    -------
-    Path
+    Handle input and output paths.
+
     """
 
-    if test:
-        filePrefix = Path("tests/resources/")
-    else:
-        filePrefix = Path(cfg.get("dataLocation")).joinpath(parentDir)
-    return filePrefix.joinpath(fileName).absolute()
+    def __init__(self):
+        """
+        IOHandler init.
+
+        """
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug("Init IOHandler")
+
+        self.output_path = None
+        self.data_path = None
+        self.model_path = None
+
+    def setPaths(self, output_path=None, data_path=None, model_path=None):
```

Comment:
I am really not sure here: putting time into changing these to camelCase, just to move it later back to snake_case?

For these particular variables, I would also like to keep consistency between the command line parameter `--output_path`, the parser variable `output_path` and the one here. It otherwise will be very confusion!

Reply:
I answer about the case point below, but if we decide to stick with camelCase for now, the consistency between the command line parameters and the internal variables are less important than the consistency with the rest of the code in my opinion.